browser/autocomplete/search_suggestions_query_check_utils.cc

```diff
@@ -0,0 +1,1772 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/autocomplete/search_suggestions_query_check_utils.h"
+
+#include <cmath>
+#include <string>
+#include <vector>
+
+#include "base/no_destructor.h"
+#include "base/notreached.h"
+#include "base/strings/strcat.h"
+#include "base/strings/string_split.h"
+#include "content/public/common/url_constants.h"
+#include "third_party/re2/src/re2/re2.h"
+#include "url/gurl.h"
+
+namespace {
+
+inline constexpr double kProbHashThreshold = 0.015;
+
+inline constexpr double kProbHashLogM[62][62] = {
```

Comment:
this should go in a separate file at least, it makes this file very difficult to navigate

Reply:
moved.